src/components/ButtonGroup.tsx

```diff
@@ -0,0 +1,23 @@
+import "./ButtonGroup.css";
+import add_icon from "../assets/svg/add.svg";
+
+function ButtonGroup({ buttonList }: any) {
+  return (
+    <div className="button_area">
+      <div className="button_list">
+        <button className="button_folder">전체</button>
+        {buttonList.map((button: any) => (
+          <button key={button.id} type="button" className="button_folder">
+            {button.name}
+          </button>
+        ))}
+      </div>
+      <button className="button_add_folder">
+        폴더 추가
+        <img className="img_add" src={add_icon} alt="" />
+      </button>
+    </div>
+  );
+}
```

Comment:
getButtonList API 요청 함수에 응답 타입이 없어서 생긴 이슈에요. 예를들어, 현재 사용중인 id, name값만 가정하게 예시를 작성해보면 아래처럼 해볼 수 있겠죠?

```suggestion
type Button = {
  id: string;
  name: string;
};

type ButtonGroupProps = {
  buttonList: Button[];
};

function ButtonGroup({ buttonList }: ButtonGroupProps) {
  if (!Array.isArray(buttonList)) {
    return null; // 또는 오류 메시지 또는 대체 UI 렌더링
  }

  return (
    <div className="button_area">
      <div className="button_list">
        <button className="button_folder">전체</button>
        {buttonList.map((button) => (
          <button key={button.id} type="button" className="button_folder">
            {button.name}
          </button>
        ))}
      </div>
      <button className="button_add_folder">
        폴더 추가
        <img className="img_add" src={add_icon} alt="" />
      </button>
    </div>
  );
}
```